internal/api/handlers/auth_handler.go

```diff
@@ -0,0 +1,140 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/Businge931/sba-api-gateway/internal/app/domain"
+	"github.com/Businge931/sba-api-gateway/internal/app/service"
+	"github.com/Businge931/sba-api-gateway/proto"
+)
+
+type AuthHandler struct {
+	authService service.AuthService
```

Comment:
Import direction is okay per the hexagonal architecture principles, but this violates the interface segregation principle, so just define this here instead of importing